contrib_bots/lib/giphy/giphy.py

```
@@ -0,0 +1,128 @@
+            Users should preface keywords with "@giphy" or the Giphy-bot @mention.
+            The bot responds also to private messages.
+            '''
+
+    def triage_message(self, message, client):
+        # To prevent infinite loop in private message, bot will detect
+        # if the sender name is the bot name it will return false.
+        if message['type'] == 'private':
+            return client.full_name != message['sender_full_name']
+
+        # Return True if we want to (possibly) response to this message.
+        original_content = message['content']
+        is_giphy_called = (original_content.startswith('@giphy ') or
+                           message['is_mentioned'])
+
+        return is_giphy_called
+
+    def handle_message(self, message, client, state_handler):
+        bot_response = get_bot_giphy_response(message, client)
+
+        if message['type'] == 'private':
+            client.send_message(dict(
+                type='private',
+                to=message['sender_email'],
+                content=bot_response,
+            ))
+        else:
+            client.send_message(dict(
+                type='stream',
+                to=message['display_recipient'],
+                subject=message['subject'],
+                content=bot_response,
+            ))
+
+
+def get_url_gif_giphy(keyword, api_key):
+    # Return a URL for a Giphy GIF based on keywords given.
+    # In case of error, e.g. failure to fetch a GIF URL, it will
+    # return a number.
+    query = {'s': keyword,
+             'api_key': api_key}
+    try:
+        data = requests.get('http://api.giphy.com/v1/gifs/translate', params=query)
+    except requests.exceptions.ConnectionError as e:  # Usually triggered by bad connection.
+        logging.warning(e)
+        return 1
+
+    search_status = data.json()['meta']['status']
+    if search_status != 200 or data.status_code != 200:
+        return 1
```

Instead of returning numbers here and in the next sanity check, you could raise exceptions which would make it more readable in the place where these error may happen.